arrow/src/main/java/org/apache/calcite/adapter/arrow/ArrowTable.java

```diff
@@ -197,6 +200,26 @@ private static TreeNode makeLiteralNode(String literal, String type) {
     case "string":
       return TreeBuilder.makeStringLiteral(literal.substring(1, literal.length() - 1));
     default:
+      return makeLiteralNode2(literal, type);
+    }
+  }
+
+  /**
+   * Creates a literal node for a data type with precision.
+   *
+   * @param literal the string representation of the value
+   * @param type the data type, expected to be in the format "type(precision,scale)"
+   * @return the corresponding TreeNode for the literal
+   */
+  private static TreeNode makeLiteralNode2(
```

Comment:
Why pass the type as a string and then parse it rather than using RelDataType?

Reply:
It may be a problem of personal coding habits. I have unified it in one function.